### PR TITLE
Milestone 4

### PR DESCRIPTION
**Pull request by Oitur**
Added like and comment models, controllers and tests

### CODE REVIEW

File: app/controllers/likes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class LikesController < ApplicationController
+  before_action :authenticate_user!
+  def create
+    post_id = like_params[:post_id]
+    current_user.likes.create(likeable: Post.find(post_id)) unless current_user.likes.find_by(likeable_id: post_id)
```
Comment (ebukaume) [inline app/controllers/likes_controller.rb]: NO application logic allowed in the views!

File: app/views/posts/_UI_new.html.erb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for(Post.new) do |f| %>
```
Comment (ebukaume) [inline app/views/posts/_UI_new.html.erb]: You should do this inside the controller, assign it to an instance variable which you then use inside your view

`@post = Post.new`

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1,9 +1,3 @@
...
-<% @posts.each do |post| %>
-  <%= post.content %>
-  <br>
-  <%= poster(post) %>
-  <h6><%= post.created_at %></h6>
-  <hr>
-<% end %>
\ No newline at end of file
+<%= render  'posts/UI_new' %>
+<%= render partial: "posts/show", collection: @posts.order('created_at DESC'), as: :post %>
```
Comment (ebukaume) [inline app/views/posts/index.html.erb]: Move this to the model. Make it a scope or a class method.

File: app/views/posts/_UI_new.html.erb (hunk 2 above)
Comment (Oitur) [reply app/views/posts/_UI_new.html.erb]: done

File: app/views/posts/index.html.erb (hunk 3 above)
Comment (Oitur) [reply app/views/posts/index.html.erb]: done

File: app/controllers/likes_controller.rb (hunk 1 above)
Comment (Oitur) [reply app/controllers/likes_controller.rb]: I'm confused on what's needed here, because this is a controller, not a view.

Comment (ebukaume) [reply app/controllers/likes_controller.rb]: I am so sorry, I made reference to the wrong file. I meant to reference the like partial

```
<div class="text-primary">
<%= pluralize(locals[:post].likes.count, 'Like') %>
</div>
<% unless like = current_user.likes.find_by(likeable_id: locals[:post].id) %>
```

Comment (Oitur) [reply app/controllers/likes_controller.rb]: done

File: app/helpers/application_helper.rb [hunk 4]
```diff
@@ -1,4 +1,7 @@
 # frozen_string_literal: true
 
 module ApplicationHelper
+  def poster(post)
+    User.find(post.user_id).name
```
Comment (maelfosso) [inline app/helpers/application_helper.rb]: I think that only `post.user.name` will be OK

**Pull request merged**